just_curious_game.ex

```diff
@@ -0,0 +1,325 @@
+defimpl String.Chars, for: JustCurious.Game do
+  @doc "string interpolated `game` prints the channel"
+  @spec to_string(struct) :: String.t
+  def to_string(game), do: game.channel
+end
+
+defmodule JustCurious.Game do
+  import Exredis.Api
+  import JustCurious.Helpers, only: [
+    redis_list_from_struct: 1,
+    string_map_to_atom_map: 1,
+    jc_bot_id: 1
+  ]
+
+  @enforce_keys ~w(name team)a
+  defstruct ~w(
+    created_at name team channel state round chainlink
+    start_timeout
+    start_reminder
+    start_reattempts
+    submitted_timeout
+    submitted_reminder
+    nominated_timeout
+    nominated_reminder
+    nominated_reattempts
+    answered_timeout
+    answered_reminder
+    answered_reattempts
+  )a
+
+  @moduledoc """
+  A `JustCurious.Game` is a FSM made up that transition through `round`s that to handle game state changes.
+
+  ## `transition`s
+  1. start
+  2. submitted
+  3. nominated
+  4. answered
+  5. finish
+
+  ## `round`s
+  1. submitting
+  2. nominating
+  3. answering
+  4. complete
+
+  ## `state`s
+  1. new    - start
+  2. active - submitted, nominated, answered
+  3. dead   - finish
+
+  ## Anatomy of the Game
+  Game state is tracked via a disposable Redis structure for preferences, participants, submissions, nominations, and answers
+
+  ``` plaintext
+  TXXX = Team, UXXX = User, 000.00X = Timestamp/Message UUID
+
+  team:T123:game:some-game
+    :players -> ["U111", "U222"]
+    :spectators -> ["U555"]
+    :invites -> ["U666"]
+    :joins -> ["U333"]
+    :name -> "Serene Desert"
+    :state -> "new"
+    :round -> "submitting"
+    :invitiation_messages -> [000.001] # Keep track of invitation messages for updating
+    :participant
+           :U111 # hash
+                :prompting -> "answer_confirmation" # prompt state tracking
+                :unconfirmed_submission -> "What's your favorite color?"
+                :submission -> "What's your favorite color?"
+                :answer -> "Green"
+                ...
+
+    :submissions # sorted set
+                :question-o-matic-1 -> 2 # a question-o-matic submitter with score
+                :U111               -> 1 # or real user submitter
+                :U222               -> 0
+
+    :nominatables # hash
+                 :000.001 -> U111 # nominatable message id and user id
+                 :000.002 -> U222
+    :nominations
+                :upvotes
+                          :U111
+                               :target -> "000.002" # user 1 upvoted second nominatable
+                               :reaction -> "heart" # with a heart reaction
+                          :U222
+                               :target -> "000.001"
+                               :reaction -> "+1" # with a heart reaction
+                :downvotes
+                          :U111 -> "000.003"
+
+  ```
+  """
+
+  @spec create(String.t, Map.t) :: %JustCurious.Game{}
+  def create(team, opts \\ %{}) do
+    name       = generate_name(team, :game)
```

Comment:
Spacing out variables like this is, perhaps unfortunately, no longer supported when using standard elixir formatter, but it looks nice!